### ISSUE TITLE
Microsoft Xbox USB dongle fails

### ISSUE THREAD

**Issue by CorentinWicht** [ChimeraOS/chimeraos#1020]
Dear Community,

As advised in other threads (#861 & #461), I have recently bought the official Microsoft Xbox USB dongle for my Xbox One Series S|X controller because the bluetooth connection was too unstable.

I was able to pair my device with the dongle on my Windows 10 computer but it failed in ChimeraOS 46...

This is what I get when I run `sudo dmesg | grep xone` at startup:
```
[    9.521780] Modules linked in: wl(POE+) zenergy(OE) amd_atl intel_rapl_msr intel_rapl_common btusb btrtl btintel btbcm btmtk bluetooth hid_generic rtw89_8852ce rtw89_8852c kvm_amd xone_dongle(OE+) xone_gip(OE) ecdh_generic usbhid rtw89_pci snd_hda_codec_hdmi kvm rtw89_core snd_hda_intel snd_intel_dspcfg crct10dif_pclmul snd_intel_sdw_acpi polyval_clmulni mac80211 polyval_generic snd_hda_codec gf128mul ghash_clmulni_intel libarc4 snd_hda_core sha1_ssse3 snd_hwdep r8169 rapl snd_pcm gigabyte_wmi wmi_bmof cfg80211 snd_timer pcspkr realtek ccp snd k10temp i2c_piix4 mdio_devres soundcore rfkill libphy mac_hid uinput crypto_user acpi_call(OE) loop nfnetlink ip_tables x_tables amdgpu btrfs blake2b_generic xor amdxcp raid6_pq i2c_algo_bit drm_exec libcrc32c gpu_sched drm_suballoc_helper drm_buddy crc32_pclmul drm_ttm_helper aesni_intel ttm nvme crypto_simd drm_display_helper xhci_pci nvme_core cryptd xhci_pci_renesas cec video nvme_auth wmi overlay sha512_ssse3 sha256_ssse3 ext4 crc32c_generic crc32c_intel crc16 mbcache jbd2
[    9.590426] xone-dongle 1-2:1.0: xone_mt76_load_firmware: resetting firmware...
[    9.613458] xone-dongle 1-2:1.0: xone_mt76_init_radio: id=0x7612
[    9.859411] xone-dongle 1-2:1.0: xone_mt76_init_address: address=62:45:b4:e7:a4:ef
[    9.895448] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=1, power=32
[    9.910451] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=6, power=32
[    9.925451] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=11, power=32
[    9.940450] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=36, power=29
[    9.955451] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=40, power=29
[    9.970456] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=44, power=29
[    9.985451] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=48, power=29
[   10.000453] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=149, power=29
[   10.015451] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=153, power=29
[   10.030452] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=157, power=29
[   10.045451] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=161, power=29
[   10.060452] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=165, power=29
[   10.063410] xone-dongle 1-2:1.0: xone_mt76_init_channels: channel=165
[   11.136521] usbcore: registered new interface driver xone-dongle
```

And when I try to press on the dongle pairing button:

```
[    9.521780] Modules linked in: wl(POE+) zenergy(OE) amd_atl intel_rapl_msr intel_rapl_common btusb btrtl btintel btbcm btmtk bluetooth hid_generic rtw89_8852ce rtw89_8852c kvm_amd xone_dongle(OE+) xone_gip(OE) ecdh_generic usbhid rtw89_pci snd_hda_codec_hdmi kvm rtw89_core snd_hda_intel snd_intel_dspcfg crct10dif_pclmul snd_intel_sdw_acpi polyval_clmulni mac80211 polyval_generic snd_hda_codec gf128mul ghash_clmulni_intel libarc4 snd_hda_core sha1_ssse3 snd_hwdep r8169 rapl snd_pcm gigabyte_wmi wmi_bmof cfg80211 snd_timer pcspkr realtek ccp snd k10temp i2c_piix4 mdio_devres soundcore rfkill libphy mac_hid uinput crypto_user acpi_call(OE) loop nfnetlink ip_tables x_tables amdgpu btrfs blake2b_generic xor amdxcp raid6_pq i2c_algo_bit drm_exec libcrc32c gpu_sched drm_suballoc_helper drm_buddy crc32_pclmul drm_ttm_helper aesni_intel ttm nvme crypto_simd drm_display_helper xhci_pci nvme_core cryptd xhci_pci_renesas cec video nvme_auth wmi overlay sha512_ssse3 sha256_ssse3 ext4 crc32c_generic crc32c_intel crc16 mbcache jbd2
[    9.590426] xone-dongle 1-2:1.0: xone_mt76_load_firmware: resetting firmware...
[    9.613458] xone-dongle 1-2:1.0: xone_mt76_init_radio: id=0x7612
[    9.859411] xone-dongle 1-2:1.0: xone_mt76_init_address: address=62:45:b4:e7:a4:ef
[    9.895448] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=1, power=32
[    9.910451] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=6, power=32
[    9.925451] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=11, power=32
[    9.940450] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=36, power=29
[    9.955451] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=40, power=29
[    9.970456] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=44, power=29
[    9.985451] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=48, power=29
[   10.000453] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=149, power=29
[   10.015451] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=153, power=29
[   10.030452] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=157, power=29
[   10.045451] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=161, power=29
[   10.060452] xone-dongle 1-2:1.0: xone_mt76_evaluate_channels: channel=165, power=29
[   10.063410] xone-dongle 1-2:1.0: xone_mt76_init_channels: channel=165
[   11.136521] usbcore: registered new interface driver xone-dongle
[   86.102996] xone-dongle 1-2:1.0: xone_mt76_suspend_radio: suspended
```

This message seems to be critical (_to be the best of my knowledge_): `xone-dongle 1-2:1.0: xone_mt76_suspend_radio: suspended`

Any idea what is happening?

Best,

C.

**Comment by pastaq** [comment]
Sometimes the dongle needs to be removed and reinserted after startup or a suspend/resume cycle to work properly.

**Comment by CorentinWicht** [comment]
> Sometimes the dongle needs to be removed and reinserted after startup or a suspend/resume cycle to work properly.

I tried that a few times already, also tried all my USB ports (front and back) and it doesn't change anything..

**Comment by CorentinWicht** [comment]
@pastaq Anything I can provide you with (i.e., logs, screenshots, etc.) to help debug that issue? I bought the official dongle specifically because the bluetooth connection was unreliable...

**Comment by alkazar** [comment]
You might have to post on the xone driver GitHub. We are not the developers of that software. However, I can tell you that I have been using the official dongle for years and it is working for me, for what that is worth.

**Comment by alkazar** [comment]
Is this the official dongle or some kind of a knock off?

**Comment by CorentinWicht** [comment]
> You might have to post on the xone driver GitHub. We are not the developers of that software. However, I can tell you that I have been using the official dongle for years and it is working for me, for what that is worth.

Thanks very much for the advice, I will then try to reach them through their Discord.

> Is this the official dongle or some kind of a knock off?

I believe it is the official one, though I ordered it from [AliExpress](https://a.aliexpress.com/_EGpSpw7) as recommended by this comment https://github.com/ChimeraOS/chimeraos/issues/969#issuecomment-2267445676.

**Comment by CorentinWicht** [comment]
Seems as though I am not the only one with the same issue (from Xone Discord):
![image](https://github.com/user-attachments/assets/4ea15f67-d040-4d8e-adb6-adac6478f55f)

**Comment by CorentinWicht** [comment]
After reading tons of advices over Reddit, I have decided to buy the [8bitdo Ultimate C 2.4G](https://www.digitec.ch/fr/s1/product/8bitdo-ultimate-c-24g-vert-android-windows-manette-35665764?supplier=406802) controller and it works out-of-the-box.

As such, I really believe that there is a driver issue with the Microsoft dongle (be it official or not).

Overall, it would be great if you could provide a list of supporter controllers (in the same vein as you do with handheld devices on your official website).

Best,

C.

**Comment by evilwombat** [comment]
Ran into the same exact problem, with the same exact VID/PID that you have. Pull the firmware file out of this version of the driver and it will work:
https://catalog.update.microsoft.com/Search.aspx?q=f1e84411-ce43-4590-8e3a-87d2fed27286

**Comment by CorentinWicht** [comment]
> Ran into the same exact problem, with the same exact VID/PID that you have. Pull the firmware file out of this version of the driver and it will work: https://catalog.update.microsoft.com/Search.aspx?q=f1e84411-ce43-4590-8e3a-87d2fed27286

Thanks for the suggestion ! Could you explain where to place that firmware file in ChimeraOS (i.e. in which folder) ? Will it be erased when updating ChimeraOS ?

Best,

C.

**Comment by evilwombat** [comment]
@CorentinWicht by the way, would you mind pointing me to where you bought your wireless adapter? I have a feeling we both got the same Chinese knock-off, because your MAC address also matches mine. Suspect there's one factory churning these out with the same exact serial number / MAC address. As long as we're not in radio range, we should be fine, but that's something to investigate another day....

**Comment by CorentinWicht** [comment]
> [@CorentinWicht](https://github.com/CorentinWicht) by the way, would you mind pointing me to where you bought your wireless adapter? I have a feeling we both got the same Chinese knock-off, because your MAC address also matches mine. Suspect there's one factory churning these out with the same exact serial number / MAC address. As long as we're not in radio range, we should be fine, but that's something to investigate another day....

Thanks for support, I will try what you recommended! 

This is the dongle I got from Aliexpress:

![Image](https://github.com/user-attachments/assets/bc27007b-65d7-4021-83d3-dff8cae0bd51)

Apparently, the item does not exist anymore (definitely a Chinese knock-off):
https://www.aliexpress.com/item/1005006560290713.html?spm=a2g0o.order_detail.order_detail_item.3.61f1f19cfyPE09

**Comment by evilwombat** [comment]
Sweet! If you feel comfortable building your own driver, I would actually recommend jumping on the latest `master` from https://github.com/dlundqvist/xone (and make sure you re-run the firmware download script). That should automatically take care of everything. If not, replacing the firmware as described earlier should also work.

**Comment by alkazar** [comment]
> Unfortunately, aince ChimeraOS is immutable I had to unlock it with `frzr-unlock` in order to install the `cabextract` package through `pacman`, would there be any other way to do it without unlocking the system [@alkazar](https://github.com/alkazar) ?


No, there is not. Cabextract is not really necessary (you can always download and extract the firmware on another computer), but you do need to unlock the system to install the firmware file.

Ultimately, I am not sure I understand the issue. The Xbox dongle firmware had a recent update that is required for some third-party dongles? Or is it a downgrade?

**Comment by alkazar** [comment]
Looks like it is a firmware downgrade. Perhaps Microsoft blocked support for third-party devices in a newer firmware update?

I am not sure how negatively this will affect official dongle users.

I will create an unstable OS build with this downgraded firmware to test it out.

**Comment by evilwombat** [comment]
The dongle has no persistent firmware storage, and the firmware is uploaded
to the dongle every time the dongle is plugged in. It's basically a WiFi
chip and a lot of them are like that. There are at least three hardware
variants of the dongle, with at least some of the counterfeits being clones
of the less-popular variant. All three variants have different driver
packages on microsoft.com, which come with different versions of the
firmware file. Near as I can tell, the firmware variants function the same
way in mission mode but the pairing button only works if the firmware
variant matches your hardware variant. The master version of xone now
detects the hw variant and tries to load a variant-specific firmware file,
before falling back to a default one.

**Comment by evilwombat** [comment]
No, even then counterfeit dongles work fine on windows, because windows
knows to read the hardware ID from the dongle and load the correct version
of the firmware file.

**Comment by alkazar** [comment]
Thanks for the info. Sounds like we just need to update our xone packages then.

**Comment by alkazar** [comment]
I guess this is only a problem now due to the proliferation of the knock offs?

**Comment by aru7777** [comment]
I still have to plug/unplug my official Xbox dongle often because it works randomly. Seems related to this issue? Is there a fix coming eventually?

**Comment by alkazar** [comment]
> I still have to plug/unplug my official Xbox dongle often because it works randomly. Seems related to this issue? Is there a fix coming eventually?

This is an unrelated issue.
ChimeraOS doesn't develop drivers. We just include the available drivers in the OS.

A workaround for what you describe can be found here:
https://deckfilter.app/2025/03/how-i-fixed-xbox-wireless-adapter-in-bazzite-linux/

Essentially you have to disable USB ports from being powered on while the computer is off, which allows the dongle to reset itself.
